### PR TITLE
Remove markupsafe pin

### LINKED ISSUE #5506: [CT-876] Could we also now remove our upper bound on `MarkupSafe`, which we put in place earlier this year due to incompatibility with Jinja2?
Remove our upper bound on `MarkupSafe`, which we put in place earlier this year due to incompatibility with Jinja2(#4745).  Also bump minimum requirement to match [Jinja2's requirements](https://github.com/pallets/jinja/blob/1c4066a4fad5aaeb2ac55809d1d38477cd23a0f6/setup.py#L6).

### PR DESCRIPTION
**Pull request by emmyoop**
resolves #5506 

### Description

Remove pin of MarkUpSafe after Jinja2 upgrade.

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: core/setup.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
     },
     install_requires=[
         "Jinja2==3.1.2",
-        "MarkupSafe>=0.23,<2.1",
+        "MarkupSafe>=2.0",
```
Comment (jtcohen6) [inline core/setup.py:52]: Any sense in just removing it entirely? I don't think we had it here until we needed to add the upper boundary earlier this year (https://github.com/dbt-labs/dbt-core/pull/4746)

Comment (emmyoop) [reply core/setup.py]: Removing the pin entirely is the right move.  I wasn't aware of the history there.

It looks like dbt-labs/dbt-server([#48](https://github.com/dbt-labs/dbt-server/pull/48)) and dbt-labs/dbt-cloud ([#4928](https://github.com/dbt-labs/dbt-cloud/pull/4928)) both pinned it as well and can not remove their pins.

**Review by jtcohen6 — APPROVED** [review]

**Comment by MichaelTiemannOSC** [comment]
This missed the 1.2.1 dbt-core cutoff.  Would be so nice if it could land in a near-term 1.2.2.

**Comment by emmyoop** [comment]
@MichaelTiemannOSC unfortunately we will not be able to remove this pin for dbt versions previous to 1.3.0.  

In dbt-core 1.3.0 we were able to upgrade Jinja2 to version 3 which is what allowed the removal of the Markupsafe pin. You can see more information on the dependency conflicts between markupsafe and Jinja2 in the [related issue](https://github.com/dbt-labs/dbt-core/issues/4745).

**Pull request merged**